domain/consensus/database/binaryserialization/common.go

```diff
@@ -0,0 +1,5 @@
+package binaryserialization
+
+import "encoding/binary"
+
+var byteOrder = binary.LittleEndian
```

Comment:
Just this can hurt performance by a lot. (the compiler can't pre-know that it's little endian because `var` is a global variable that can change in runtime, so it has to jump to the variable, so it can't inline the little endian, so it can't use the fact that the computer is also little endian and optimize this out)